Source/ImPlotWidgets/Private/LineFilledGraphWidget.cpp

```diff
@@ -44,6 +45,14 @@ void SFilledGraphWidget::OnImGuiDraw() const
 			XAxisValues.Add(XValue);
 		}
 
+		Algo::Sort(InData2D, [](const FDataPoint& LeftDataPoint, const FDataPoint& RightDataPoint) {return LeftDataPoint.X < RightDataPoint.X; });
```

Comment:
```suggestion
		Algo::Sort(InData2D, [](const FDataPoint& LeftDataPoint, const FDataPoint& RightDataPoint)
		{
			return LeftDataPoint.X < RightDataPoint.X; 
		});
```

Reply:
Should line 48-55 actually go into the Construct() function? Right now we take a copy of the data TArray in the Slate args. So when the widget has been constructed, the plot data cannot change. Therefore we don't need to sort the data and separate it into the X and Y arrays in OnImGuiDraw(), which I assume is called on Tick()?

Now an extension to this (for later) would be to take a pointer to an array as the argument. Then we could live update the data, and live update the graph. Probably, it might not 'just work'.

Reply:
Yeah OnImGuiDraw() is called on Tick() so this would definitely make sense, I didn't even come to think of it. I also had to change it to be non const to be able to do the sort, and moving that part over would let me make it const again.